pkg/parse/source.go

```diff
@@ -96,6 +96,9 @@ func (o *files) readConfigFiles(state *sourceState) status.Error {
 	if err != nil {
 		return status.PathWrapError(errors.Wrap(err, "listing files in the configs directory"), syncDir.OSPath())
 	}
+	if state.commit != commit {
```

Comment:
The commit passed in the `readConfigFiles` function is always the same as `state.commit`. You need to recompute the symlink.

Reply:
Done, addressed in an earlier reply, now passing the parser with its hydratedRoot and hydratedLink to compute the new commit if changed.